Orange/widgets/evaluate/owrocanalysis.py

```diff
@@ -601,6 +606,50 @@ def _setup_plot(self):
                 warning = "All ROC curves are undefined"
         self.warning(warning)
 
+    def _on_mouse_moved(self, pos):
+        curves = self.plot.curves
+        for i in self.selected_classifiers:
+            sp = curves[i].childItems()[0]
+            act_pos = sp.mapFromScene(pos)
+            pts = sp.pointsAt(act_pos)
+
+            curve_data = self.curve_data(self.target_index, i)
+
+            if self.roc_averaging == OWROCAnalysis.Merge:
+                curve_data = curve_data.merged
+            elif self.roc_averaging == OWROCAnalysis.Vertical:
+                curve_data = curve_data.avg_vertical
+            elif OWROCAnalysis.Threshold:
+                curve_data = curve_data.avg_threshold
+            elif OWROCAnalysis.NoAveraging:
+                # currently not implemented
+                return
+
+            if len(pts) > 0:
+                mouse_pt = pts[0].pos()
+
+                if self._tooltip_cache:
+                    if numpy.linalg.norm(mouse_pt - self._tooltip_cache.pos()) < 10e-6:
+                        return
+                    else:
+                        self.plot.removeItem(self._tooltip_cache)
+                        self._tooltip_cache = None
+
+                curve_pts = curve_data.points
+
+                # Find closest point on curve and display it
+                idx_closest = numpy.argmin([numpy.linalg.norm(mouse_pt - [curve_pts.fpr[idx], curve_pts.tpr[idx]])
+                                            for idx in range(len(curve_pts.thresholds))])
+
+                thresh = curve_pts.thresholds[idx_closest]
+                if not numpy.isnan(thresh):
+                    item = pg.TextItem(text="{:.3f}".format(thresh), color=(0, 0, 0))
```

Comment:
Why not `QToolTip.showText`?

Reply:
Mainly because I am new to QT and was just following a few resources online.  
I've fixed most of the things that you've mentioned and some other bugs.  
Thank you for the helpful comments!